README.md

```diff
@@ -100,6 +100,7 @@ Finally, enable all of the rules that you would like to use.  Use [our preset](#
 * [react/no-redundant-should-component-update](docs/rules/no-redundant-should-component-update.md): Prevent usage of `shouldComponentUpdate` when extending React.PureComponent
 * [react/no-render-return-value](docs/rules/no-render-return-value.md): Prevent usage of the return value of `React.render`
 * [react/no-set-state](docs/rules/no-set-state.md): Prevent usage of `setState`
+* [react/no-static-typos](docs/rules/no-static-typos.md): Prevent casing typos when declaring static class properties.
```

Comment:
> as I never remember if it was prevProps, nextProps, prevState or nextState

To be clear, my suggestion is to apply this rule to lifecycle method names, not their arguments. e.g. `componentWillMount`. A rule that enforces naming of lifecycle method arguments would be interesting, but I think that is definitely a different rule than typo detection.